aptos-move/aptos-vm/src/verifier/event_validation.rs

```diff
@@ -0,0 +1,149 @@
+                    let func_handle = module.function_handle_at(func_instantiation.handle);
+                    let module_handle = module.module_handle_at(func_handle.module);
+                    let module_addr = module.address_identifier_at(module_handle.address);
+                    let module_name = module.identifier_at(module_handle.name);
+                    let func_name = module.identifier_at(func_handle.name);
+                    if module_addr
+                        != &AccountAddress::from_hex_literal("0x1").expect("valid address")
+                        || module_name.as_str() != EVENT_MODULE_NAME
+                        || func_name.as_str() != EVENT_EMIT_FUNCTION_NAME
+                    {
+                        continue;
+                    }
+                    let param = module
+                        .signature_at(func_instantiation.type_parameters)
+                        .0
+                        .first()
+                        .ok_or_else(|| {
+                            metadata_validation_error(
+                                "Missing parameter for 0x1::event::emit function",
+                            )
+                        })?;
+                    match param {
+                        StructInstantiation(index, _) | Struct(index) => {
+                            let struct_handle = &module.struct_handle_at(*index);
+                            let struct_name = module.identifier_at(struct_handle.name);
+                            if struct_handle.module != module.self_handle_idx() {
+                                metadata_validation_err(format!("{} passed to 0x1::event::emit function is not defined in the same module", struct_name).as_str())
+                            } else if !event_structs.contains(struct_name.as_str()) {
+                                metadata_validation_err(format!("#[event] attribute is missing for {} passed to 0x1::event::emit function", struct_name).as_str())
+                            } else {
+                                Ok(())
+                            }
+                        },
+                        _ => metadata_validation_err(
+                            "Reference to non-struct parameter for 0x1::event::emit function",
+                        ),
+                    }?;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Given a module id extract all event metadata
+pub(crate) fn extract_event_metadata_from_module(
+    session: &mut SessionExt,
+    module_id: &ModuleId,
+) -> VMResult<HashSet<String>> {
+    let metadata = session.load_module(module_id).map(|module| {
+        CompiledModule::deserialize(&module)
+            .map(|module| aptos_framework::get_metadata_from_compiled_module(&module))
+    });
+
+    if let Ok(Ok(Some(metadata))) = metadata {
+        extract_event_metadata(&metadata)
+    } else {
+        Ok(HashSet::new())
+    }
```

Comment:
shouldn't we fail if we can't load the module or can't deserialize it? granted I don't see how we get there, but ... it is a bit concerning we're discarding errors in line 129

Reply:
I think I just copied what resource_group does in this case without thinking through it too much. If we wanna change, I think we should change both places but it may has replay issues.